frontend/src/components/SamplesList.tsx

```diff
@@ -11,99 +11,38 @@ import { DownloadModal } from "./DownloadModal";
 import "ag-grid-enterprise";
-import { CellValueChangedEvent } from "ag-grid-community";
-import { parseSearchQueries } from "../utils/parseSearchQueries";
+import { CellValueChangedEvent, ColDef } from "ag-grid-community";
 import { ErrorMessage, LoadingSpinner, Toolbar } from "../shared/tableElements";
 
 const POLLING_INTERVAL = 2000;
 const max_rows = 500;
 
 interface ISampleListProps {
+  columnDefs: ColDef[];
+  defaultColDef: ColDef;
+  getRowData: (samples: Sample[]) => any[];
   height: number;
   setUnsavedChanges?: (val: boolean) => void;
-  searchVariables?: SampleMetadataWhere;
+  searchVariables?: SampleWhere;
+  filter: (searchVal: string) => SampleWhere;
   exportFileName?: string;
-  sampleQueryParamFieldName?: string;
-  sampleQueryParamValue?: string;
-}
-
-function sampleFilterWhereVariables(
-  uniqueQueries: string[]
-): SampleMetadataWhere[] {
-  if (uniqueQueries.length > 1) {
-    return [
-      { cmoSampleName_IN: uniqueQueries },
-      { importDate_IN: uniqueQueries },
-      { investigatorSampleId_IN: uniqueQueries },
-      { primaryId_IN: uniqueQueries },
-      { sampleClass_IN: uniqueQueries },
-      { cmoPatientId_IN: uniqueQueries },
-      { cmoSampleIdFields_IN: uniqueQueries },
-      { sampleName_IN: uniqueQueries },
-      { preservation_IN: uniqueQueries },
-      { tumorOrNormal_IN: uniqueQueries },
-      { oncotreeCode_IN: uniqueQueries },
-      { collectionYear_IN: uniqueQueries },
-      { sampleOrigin_IN: uniqueQueries },
-      { tissueLocation_IN: uniqueQueries },
-      { sex_IN: uniqueQueries },
-      { libraries_IN: uniqueQueries },
-      { sampleType_IN: uniqueQueries },
-      { species_IN: uniqueQueries },
-      { genePanel_IN: uniqueQueries },
-    ];
-  } else {
-    return [
-      { cmoSampleName_CONTAINS: uniqueQueries[0] },
-      { importDate_CONTAINS: uniqueQueries[0] },
-      { investigatorSampleId_CONTAINS: uniqueQueries[0] },
-      { primaryId_CONTAINS: uniqueQueries[0] },
-      { sampleClass_CONTAINS: uniqueQueries[0] },
-      { cmoPatientId_CONTAINS: uniqueQueries[0] },
-      { cmoSampleIdFields_CONTAINS: uniqueQueries[0] },
-      { sampleName_CONTAINS: uniqueQueries[0] },
-      { preservation_CONTAINS: uniqueQueries[0] },
-      { tumorOrNormal_CONTAINS: uniqueQueries[0] },
-      { oncotreeCode_CONTAINS: uniqueQueries[0] },
-      { collectionYear_CONTAINS: uniqueQueries[0] },
-      { sampleOrigin_CONTAINS: uniqueQueries[0] },
-      { tissueLocation_CONTAINS: uniqueQueries[0] },
-      { sex_CONTAINS: uniqueQueries[0] },
-      { libraries_CONTAINS: uniqueQueries[0] },
-      { sampleType_CONTAINS: uniqueQueries[0] },
-      { species_CONTAINS: uniqueQueries[0] },
-      { genePanel_CONTAINS: uniqueQueries[0] },
-    ];
-  }
-}
-
-function getSampleMetadata(samples: Sample[]) {
-  return samples.map((s: any) => {
-    return {
-      ...s.hasMetadataSampleMetadata[0],
-      revisable: s.revisable,
-    };
-  });
```

Comment:
These lines were simply moved to `frontend/src/shared/helpers.tsx`. These lines no longer belong in this component because they're specifically used for working with samples' metadata. Moving them out lets us keep this component generic so that it can also be used for displaying samples' Tempo event data on the Cohorts page.